crates/forge/bin/cmd/test/mod.rs

```diff
@@ -655,6 +668,83 @@ impl TestArgs {
                         }
                     }
                 }
+
+                // Collect and merge gas snapshots.
+                for (group, new_snapshots) in result.gas_snapshots.iter() {
+                    gas_snapshots.entry(group.clone()).or_default().extend(new_snapshots.clone());
+                }
+            }
+
+            // Write gas snapshots to disk if any were collected.
+            if !gas_snapshots.is_empty() {
+                // Check for differences in gas snapshots if `FORGE_SNAPSHOT_CHECK` is set.
+                // Exiting early with code 1 if differences are found.
+                if std::env::var("FORGE_SNAPSHOT_CHECK").is_ok() {
+                    let differences_found = gas_snapshots.clone().into_iter().fold(
```

Comment:
Good point, I think this could be added as a future optimization, to only write snapshots that differ. Combine this with a glob over the files in the `snapshots` folder to filter out unmatched snapshots that only exist in the old and not in the new (to avoid staining).